src/extension/background-script/actions/nostr/generatePrivateKey.ts

```diff
@@ -0,0 +1,31 @@
+import * as secp256k1 from "@noble/secp256k1";
+
+import state from "../../state";
+
+const generatePrivateKey = async () => {
+  const connector = await state.getState().getConnector();
+  try {
+    const response = await connector.signMessage({
+      message:
+        "DO NOT EVER SIGN THIS TEXT WITH YOUR PRIVATE KEYS! IT IS ONLY USED FOR DERIVATION OF NOSTR KEY PAIRS, DISCLOSING THIS SIGNATURE WILL COMPROMISE YOUR NOSTR IDENTITY!",
+      key_loc: {
+        key_family: 0,
+        key_index: 0,
+      },
+    });
+    const keymaterial = response.data.signature;
+    const uint8 = Uint8Array.from(
+      keymaterial.split("").map((x) => x.charCodeAt(0))
+    );
+    const hex = Buffer.from(secp256k1.utils.hashToPrivateKey(uint8));
```

Comment:
the variable name says `hex` but the content is not hex here. 

I don't think you should do this `keymaterial.split` thing. Are you sure this is needed? (looks a bit like a stack overflow answer ;) )

can the `secp256k1.utils.hexToBytes()` help here? that takes a hex string and converts it into bytes that can be passed to `hashToPrivateKey`.